tests/unit/test_validate_input.py

```diff
@@ -0,0 +1,29 @@
+import os
+import sys
+import tempfile
+import unittest
+
+from idseq_dag.steps.run_validate_input import PipelineStepRunValidateInput
+from idseq_dag.exceptions import InsufficientReadsError, InvalidFileFormatError
+
+fasta_filename = os.path.join(os.path.dirname(__file__), "fixtures", "reads.fasta")
+fastq_filename = os.path.join(os.path.dirname(__file__), "fixtures", "reads.fastq")
+
+class TestValidateInput(unittest.TestCase):
+    def test_validate_input(self):
+        PipelineStepRunValidateInput.quick_check_file(None, fasta_filename, is_fastq=False, max_fragments_to_check=100)
+        PipelineStepRunValidateInput.quick_check_file(None, fastq_filename, is_fastq=True, max_fragments_to_check=100)
+        with self.assertRaises(FileNotFoundError):
+            PipelineStepRunValidateInput.quick_check_file(None, "nonexistent", is_fastq=True, max_fragments_to_check=100)
+        with tempfile.NamedTemporaryFile() as tf:
+            with self.assertRaises(InsufficientReadsError):
```

Comment:
We should have different test functions for each test... Any performance reason not to do so?

Reply:
No performance reason, but I dislike doing that. It's overly verbose. Any line in this test failing indicates an issue, and it's fairly self evident what the issue is. I could rewrite this as separate tests, but it would make it 2-3 times as long for no semantic benefit.

Reply:
We strongly disagree on this.
In general, because it is just helps as the number of tests grow.
In this particular case, because future people looking at this are not necessarily as familiar as you with it and do not really know this function and don't know what particular types of format errors you are exactly trying to test.

Anyway, just an opinion, so, up to you.

Reply:
Added comments to clarify the purpose of each test assertion.

I definitely have a very strong bias toward brevity while preserving readability. But in this specific location I was only trying to add a regression test for a bug fix that I made; there was no unit testing at all in this module before. I think writing a more comprehensive test suite for this module should be tracked as a separate task.

Reply:
I do appreciate having some tests (even if packed in one) than none at all! :)